common/src/margins.rs

```diff
@@ -0,0 +1,213 @@
+use crate::{
+    dodeca::Vertex,
+    math,
+    node::VoxelData,
+    voxel_math::{ChunkDirection, CoordAxis, CoordSign, Coords, SimpleChunkOrientation},
+};
+
+/// Updates the margins of both `voxels` and `neighbor_voxels` at the side they meet at.
+/// It is assumed that `voxels` corresponds to a chunk that lies at `vertex` and that
+/// `neighbor_voxels` is at direction `direction` from `voxels`.
+pub fn fix_margins(
+    dimension: u8,
+    vertex: Vertex,
+    voxels: &mut VoxelData,
+    direction: ChunkDirection,
+    neighbor_voxels: &mut VoxelData,
+) {
+    let neighbor_orientation = match direction.sign {
```

Comment:
Would it make sense to bundle up this entire `match` as a helper method/getter, either on `Vertex` or at least freestanding in this module?

Reply:
Good question. I'm also considering bundling it into the return value of `get_chunk_neighbor`.